Amazon.QLDB.Driver/session/SessionPool.cs

```diff
@@ -64,11 +64,8 @@ public T Execute<T>(Func<TransactionExecutor, T> func, RetryPolicy retryPolicy,
                 return this.retryHandler.RetriableExecute(
                     () => session.Execute(func),
                     retryPolicy,
-                    () =>
-                    {
-                        session.Dispose();
-                        session = this.GetSession();
-                    },
+                    () => session = this.StartNewSession(),
+                    () => session = this.GetSession(),
```

Comment:
I think there's a semaphore leak here? You aren't releasing the previous ones in the retryPolicy before fetching more.

Reply:
It was released when the abort transaction failed.